rasterio/_io.pyx

```diff
@@ -1130,6 +1129,8 @@ cdef class DatasetWriterBase(DatasetReaderBase):
                 OSRImportFromProj4(osr, proj_c)
         # Fall back for CRS strings like "EPSG:3857."
         else:
+            if crs is None:
```

Comment:
It might not hurt to make the `else` in the line above an `elif isinstance(crs, str) or crs is None`, and then a final `else` that raises a`CRSError()`.  Consider this, which raises not very helpful `AttributeError`:

```python
import rasterio as rio

with rio.open('tests/data/RGB.byte.tif') as src:
    profile = src.profile.copy()

profile.update(crs=['trash'])

with rio.open('test.tif', 'w', **profile) as dst:
    pass
```